realm/realm-jni/src/io_realm_internal_TableQuery.cpp

```diff
@@ -933,15 +934,13 @@ static void TableQuery_BinaryPredicate(JNIEnv *env, jlong nativeQueryPtr, jlongA
             value2 = BinaryData();
         }
         else {
-            jbyte* bytePtr = env->GetByteArrayElements(value, NULL);
-            if (!bytePtr) {
+            if (!bytes.ptr()) {
```

Comment:
I agree, but this here http://stackoverflow.com/questions/25852147/when-does-getbytearrayelements-return-null seems to indicate it might fail in other ways. At least this check isn't harmful.